### PR TITLE
rework nss-systemd recursion lock

### PR DESCRIPTION
**Pull request by poettering**

### CODE REVIEW

File: src/basic/dlfcn-util.h [hunk 1]
```diff
@@ -0,0 +1,9 @@
+/* SPDX-License-Identifier: LGPL-2.1+ */
+#pragma once
+
+#include <dlfcn.h>
+
+static inline void dlclosep(void **dl) {
+        if (*dl)
+                dlclose(dl);
+}
```
Comment (keszybz) [inline src/basic/dlfcn-util.h]: Yeah, I think the dependency on `dl` is "free" in any realistic scenario.

File: src/basic/dlfcn-util.h [hunk 2]
```diff
@@ -0,0 +1,9 @@
+/* SPDX-License-Identifier: LGPL-2.1+ */
+#pragma once
+
+#include <dlfcn.h>
+
+static inline void dlclosep(void **dl) {
+        if (*dl)
+                dlclose(dl);
```
Comment (keszybz) [inline src/basic/dlfcn-util.h]: Dereference is missing here. Please just use `DEFINE_TRIVIAL_CLEANUP_FUNC()`.

Comment (keszybz) [reply src/basic/dlfcn-util.h]: I force-pushed with this change. But it's rather strange that CI didn't complain about this at all.

Comment (poettering) [reply src/basic/dlfcn-util.h]: oh god. So weird this never crashed on me...

Thanks for the fix...

**Comment by poettering** [comment]
thanks for the fix, lgtm, seeting green label

**Pull request merged**